Preparing for 1.4

zalun opened the pull request:
Before we will push it live we need to have ABH 1.4 downloadable on AMO
- display a warning if ABH is not responding with an object containing `then`
- set the desired ABH number to 1.4

zalun committed:
ABH is asynchronous
Before we will push it live we need to have ABH 1.4 downloadable on AMO

* display a warning if ABH is not responding with an object containing `then`
* set the desired ABH number to 1.4

seanmonstar commented on media/base/js/flightdeck/FlightDeck.js:
```diff
@@ -254,24 +253,26 @@ module.exports = new Class({
-                                    'Wrong response from Add-on Builder Helper. Please <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=573778">let us know</a>'
-                                );
-                            }
+                            var result = ABH().send("install", responseText).then(function(response) {
```
it's async now, so `result` variable isn't needed

seanmonstar commented on media/base/js/flightdeck/FlightDeck.js:
```diff
@@ -294,12 +295,15 @@ module.exports = new Class({
+        log.debug('uninstalling add-on');
+        ABH().send('uninstall').then(function(response){
+            if (response.success) {
+                // XXX: no idea why it's not working without this line
```
really? tried removing it, again? we should find out why... and if not, want to log a useful message there?

zalun replied:
the message is later in whenXpiUninstalled
I was adding and removing this line a few times as I couldn't believe it myself
I also restarted firefox althought not entire system :)

seanmonstar commented:
`mozFlightDeck` is also accessed in PackageController, line 200, to toggle the debug console.

zalun committed:
Update after code review and fix in Addon Builder Helper

* no result needed
* toggleConsole debug removed
* debug removed - it started magically working after full system restart

zalun committed:
SDK 1.4 added

zalun committed:
Added --useversion option as SDK 1.4 addon-kit and api-utils do not
specify version in the package.json

tests are run against addon-sdk-1.4

lowest approved SDK changed to 1.4 to prevent people from using 1.3

seanmonstar commented on apps/jetpack/management/commands/add_core_lib.py:
```diff
@@ -10,11 +10,17 @@ class Command(BaseCommand):
     def handle(self, sdk_dir_name, *args, **kwargs):
         #try:
-        create_or_update_SDK(sdk_dir_name, options=kwargs['options'])
+        create_or_update_SDK(sdk_dir_name,
```
you wouldnt rather change the method header?

`def handle(self, sdk_dir_name, version=None, options=None, *args, **kwargs):`

seanmonstar commented:
the comment about `handle` is more just a question. so r+!

zalun committed:
handle function refactored
